lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -8316,11 +8317,13 @@ class DecomposeAtenOneHotOp : public OpRewritePattern<AtenOneHotOp> {
     llvm::SmallVector<int64_t> unsqueezeShape(inputType.getSizes());
     unsqueezeShape.push_back(1);
     auto unsqueezeType =
-        ValueTensorType::get(context, unsqueezeShape, si64Type);
+        ValueTensorType::get(context, unsqueezeShape, inputType.getDtype());
     Value unsqueezeTensor = rewriter.create<AtenUnsqueezeOp>(
         loc, unsqueezeType, input,
         rewriter.create<ConstantIntOp>(loc,
                                        rewriter.getI64IntegerAttr(inputRank)));
```

Comment:
Optional: what would you think about replacing this highlighted section with a util function? That's like 7 lines of boilerplate code just vanishing into thin air!
```cpp
Value unsqueezeTensor = Torch::unsqueezeTensor(rewriter, op, input, -1);
```

Reply:
Coolio, didn't know that existed